File: mw-kreator-regul-biznesowych/app-modul-model/src/main/java/model/encje/RegulaEncja.java
package model.encje;

import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.Fetch;

import javax.persistence.*;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "reguly")
public class RegulaEncja extends Encja{

    private String kod;

    private String tresc;


    public RegulaEncja() {
    }

    public RegulaEncja(String kod, String tresc) {
        this.kod = kod;
        this.tresc = tresc;
    }

    @OneToMany(mappedBy = "regula", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @OrderBy("id")
    @Fetch( FetchMode.SELECT) //uwaga bez tego zduplikowane rekordy
    private List<ParametrRegulyEncja> parametry=null;


    @OneToMany(mappedBy = "regulaWolajaca", fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @Fetch( FetchMode.SELECT)
    private Set<WywolanieRegulyEncja> wywolaniaRegul=null;




    public String getKod() {
        return kod;
    }

    public void setKod(String kod) {
        this.kod = kod;
    }

    public String getTresc() {
        return tresc;
    }

    public void setTresc(String tresc) {
        this.tresc = tresc;
    }

    public List<ParametrRegulyEncja> getParametry() {
        return parametry;
    }

    public void setParametry(List<ParametrRegulyEncja> parametry) {
        this.parametry = parametry;
    }

    public Set<WywolanieRegulyEncja> getWywolaniaRegul() {
        return wywolaniaRegul;
    }

    public void setWywolaniaRegul(Set<WywolanieRegulyEncja> wywolaniaRegul) {
        this.wywolaniaRegul = wywolaniaRegul;
    }
}

File: mw-kreator-regul-biznesowych/app-modul-model/src/main/kotlin/model/encje/RegulaEncja.kt
package model.encje

import model.nlp.Sekwencja
import javax.persistence.*

@Entity
@Table(name = "reguly")
data class RegulaEncja(
        val kod: String,
        val tresc: String
) : Encja(), IWalidowalny {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @Version
    @Column(name = "WERSJA")
    var wersja: Long? = null

    @OneToMany(mappedBy = "regula", fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.ALL))
    var parametry: MutableList<ParametrRegulyEncja>? = null

    @Transient
    var sekwencja: Sekwencja? = null

    @Transient
    override fun waliduj(): List<String> {
        val bledy: MutableList<String> = mutableListOf<String>()

        if (parametry != null) {
            parametry!!.forEach {
                val wynik=it.waliduj()
                if(!wynik.isEmpty()) {
                    bledy.addAll(wynik)
                }
            }
        }

        return bledy
    }
}